esp-hal/src/uart/mod.rs

```diff
@@ -936,6 +972,16 @@ impl<'d> UartRx<'d, Blocking> {
         Ok(uart_rx)
     }
 
+    /// Waits for a break condition to be detected.
+    /// 
+    /// This is a blocking function that will continuously check for a break condition.
+    #[instability::unstable]
+    pub fn wait_for_break(&mut self) {
+        while !self.uart.info().brk_det().bit_is_set() {
+            // wait
+        }
+    }
+
```

Comment:
@bugadani - I added `wait_for_break` to complete the HIL test but I would intend to use it more as an interrupt handler now that we have Rx event for break detection.

Do you approve of this?

Reply:
I think doing both should be fine. This function has lower latency than any interrupt handler would, so it can probably stay and remain useful in this form.

It makes sense to add an `UartInterrupt::Break` variant.

It would be nice to have a `wait_for_break_with_timeout` variant, and an `async fn wait_for_break_async`, too.